third_party/xla_client/computation_client.cc

```diff
@@ -66,8 +80,31 @@ bool IsLocalDevice(
     return true;
   }
   Device device = ParseDevice(mp_device);
-  return device.ordinal == parsed_device.id &&
-         device.kind == parsed_device.type;
+  string task_device_key = BuildTaskDeviceKey(parsed_device.task, device.kind);
+  auto it = dev_task_map.find(task_device_key);
+  return it != dev_task_map.end()
+             ? (device.ordinal == it->second + parsed_device.id)
```

Comment:
Would you mind helping me understand what values `it->second` and `parsed_device.id` range from?

Reply:
`it->second` contains the minimum global device ID (for a given device kind) for the given task.